src/ImmersedBoundaries/ImmersedBoundaries.jl

```diff
@@ -137,7 +158,16 @@ const IBG = ImmersedBoundaryGrid
 Adapt.adapt_structure(to, ibg::IBG{FT, TX, TY, TZ}) where {FT, TX, TY, TZ} =
     ImmersedBoundaryGrid{TX, TY, TZ}(adapt(to, ibg.underlying_grid), adapt(to, ibg.immersed_boundary), adapt(to, ibg.active_cells_map))
 
-with_halo(halo, ibg::ImmersedBoundaryGrid) = ImmersedBoundaryGrid(with_halo(halo, ibg.underlying_grid), ibg.immersed_boundary)
+function with_halo(halo, ibg::ImmersedBoundaryGrid)
+    TX, TY, TZ = topology(ibg)
+
+    # I think we have to recompute the active cells map?
+    active_cells_map = !isnothing(ibg.active_cells_map)
```

Comment:
hm....
looking here --> https://github.com/CliMA/Oceananigans.jl/blob/19dbe19a76043100ac7fd08bdd754cc362fc19dd/src/ImmersedBoundaries/active_cells_map.jl#L29

reads to me that the active_cells_map is only about the interior? That means we don't need to recompute the active cells map? is this true? remove comment if so?

Reply:
this is indeed true